### PR TITLE
Create and update companies

### PR DESCRIPTION
**Pull request by RayyanMerchant**

### CODE REVIEW

File: placementApp/urls.py [hunk 1]
```diff
@@ -5,7 +5,8 @@
 router = routers.DefaultRouter()
 router.register("students", views.StudentViewSet)
 router.register("positions", views.PositionViewSet)
-
+router.register("company", views.CompanyViewSet)
+router.register("company/<int:pk>", views.CompanyViewSet)
```
Comment (parag1999) [inline placementApp/urls.py]: No need to add another url company/<int:pk>  router.register takes care of it by default

File: placementApp/views.py [hunk 2]
```diff
@@ -22,6 +22,20 @@ class PositionViewSet(
     queryset = Position.objects.all()
     serializer_class = PositionSerializer
 
+class CompanyViewSet(
+    mixins.CreateModelMixin, mixins.UpdateModelMixin, mixins.ListModelMixin, mixins.RetrieveModelMixin, viewsets.GenericViewSet,
+):
+    permission_classes = (permissions.AllowAny,)
+    queryset = Company.objects.all()
+    serializer_class = CompanySerializer
+    def patch(self, request, pk):
```
Comment (parag1999) [inline placementApp/views.py]: There is no need for a patch function the UpdateModelMixin handles both update and partial update through put and patch on its own so no need to define or overwrite it

File: placementApp/urls.py (hunk 1 above)
Comment (RayyanMerchant) [reply placementApp/urls.py]: I know fixed it

File: placementApp/views.py (hunk 2 above)
Comment (RayyanMerchant) [reply placementApp/views.py]: Patch request woudn't work it said Patch isn't allowed.

File: placementApp/urls.py (hunk 1 above)
Comment (preet4737) [reply placementApp/urls.py]: Rayyan directly use ModelViewSet

File: placementApp/views.py (hunk 2 above)
Comment (preet4737) [reply placementApp/views.py]: Use ModelViewSet for Company.

Comment (parag1999) [reply placementApp/views.py]: I execute the code after removing patch function and both put and patch works fine

**Pull request merged**